src/sentry/migrations/0378_backfill_perf_issue_events_issue_platform.py

```diff
@@ -0,0 +1,648 @@
+    def __call__(
+        self,
+        attempted_row: TransactionRow,
+        exc: Exception,
+        occurrence_nodestore_saved: bool,
+        occurrence_eventstream_sent: bool,
+    ) -> None:
+        pass
+
+
+def print_success(
+    timestamp: datetime, created_group: "Group", mapped_occurrence: IssueOccurrenceData
+) -> None:
+    print(  # noqa: S002
+        f"project_id={created_group.project_id}, group_id={created_group.id}, event_id={mapped_occurrence['event_id']}, occurrence_id={mapped_occurrence['id']}, fingerprint={mapped_occurrence['fingerprint']}"
+    )
+
+
+def print_exception_on_error(
+    attempted_row: TransactionRow,
+    exc: Exception,
+    occurrence_nodestore_saved: bool,
+    occurrence_eventstream_sent: bool,
+) -> None:
+    print(  # noqa: S002
+        f"row={attempted_row}, exception={exc}, nodestore_saved={occurrence_nodestore_saved}, eventstream_sent={occurrence_eventstream_sent}"
+    )
+
+
+def backfill_eventstream(apps: Any, schema_editor: Any) -> None:
+    """
+    Backfills Performance-issue events from the transaction table to the IssuePlatform dataset(search_issues).
+    1. Source Groups from postgres as the 'source of truth' for issues.
+    2. For each project with a performance issue:
+         a. Query snuba for the transactions backing a perf issue. We only need (project_id, group_id, event_id).
+            This is paginated with a limit of 10000 transactions per page to avoid pull in too much data into memory.
+         b. Query snuba for generic events backing a perf issue (we'll be dual-writing to both tables, so we need
+            to ensure the event doesn't already exist in search_issues).
+         c. For each transaction:
+              i. Retrieve the Group row from postgres.
+              ii. Retrieve the GroupHash row from postgres.
+              iii. Retrieve the entire raw Transaction from node store.
+              iv.
+              v. Map the Transaction event to an instance of IssueOccurrenceData.
+              v. Save the IssueOccurrence to node store. Normally this step would also save the appropriate models to
+                 postgres like Group, GroupHash, GroupEnvironment, GroupRelease, and increment Release counts, but
+                 we aren't doing that on this backfill since they should already exist when the transaction was already
+                 ingested through the old style of performance issue creation.
+              vi. Send the IssueOccurrence through the eventstream to be saved in search_issues.
+         c. Repeat step (a), until the paginated snuba query returns no data.
+    """
+
+    Group = apps.get_model("sentry", "Group")
+    GroupHash = apps.get_model("sentry", "GroupHash")
+
+    projects_with_perf_issues = (
+        Group.objects.filter(type__in=PERFORMANCE_TYPES)
+        .values("project_id")
+        .order_by("project_id")
+        .distinct()
+    )
+
+    for project_perf_issue in projects_with_perf_issues:
```

Comment:
It'd be good to have an idea of progress as we iterate through everything here. I think we can just wrap this with `WithProgressBar`, which is used as part of the ranged queryset wrapper usually https://github.com/getsentry/sentry/blob/3da131c31566e04cc32398ee91d901b69a260fa0/src/sentry/utils/query.py#L178